src/Form.js

```diff
@@ -15,25 +17,38 @@ const configureDate = (date) => {
 const submissionType = (e) => {
   console.log(location.pathname)
   if(location.pathname === '/') {
+    setLoading(true)
    return getCoords(e, userDate)
   } else {
    return fetchSpecific(e)
   }
 }
 
+const maxDateMaker = () => {
+  if(location.pathname === '/') {
+    setFormDate(dayjs().subtract(7, 'month').format('YYYY/MM/DD').split('/').join('-'))
+  } else {
+    setFormDate(dayjs().format('YYYY/MM/DD').split('/').join('-'))
+  }
+}
+
+useEffect(() => {
+  console.log('hello form')
+  console.log(formDate)
+  maxDateMaker()
+},[maxDateMaker, formDate])
```

Comment:
Nice job getting everything done in this one maxDateMaker function. You may want to look into if you actually need a useEffect hook here- you might be able to do this functionality in just another function.

Reply:
Thank you! My theory around the useEffect hook here was to force a re-render of the Form when the max date changes, so as not to accidentally carry over the wrong max value to the wrong page. I did initially try setting the max to a function value from Form, but this caused a few errors. Fortunately, this useEffect worked on its initial implementation!